ethash/src/compute.rs

```diff
@@ -210,17 +210,18 @@ pub fn slow_get_seedhash(block_number: u64) -> H256 {
 	SeedHashCompute::resume_compute_seedhash([0u8; 32], 0, block_number / ETHASH_EPOCH_LENGTH)
 }
 
-#[inline]
 fn fnv_hash(x: u32, y: u32) -> u32 {
 	return x.wrapping_mul(FNV_PRIME) ^ y;
 }
 
-#[inline]
 fn sha3_512(input: &[u8], output: &mut [u8]) {
 	unsafe { sha3::sha3_512(output.as_mut_ptr(), output.len(), input.as_ptr(), input.len()) };
 }
 
-#[inline]
+fn sha3_512_inplace(input: &mut [u8]) {
+	unsafe { sha3::sha3_512(input.as_mut_ptr(), input.len(), input.as_ptr(), input.len()) };
```

Comment:
with a comment explaining why it's safe? couldn't a `sha3` implementation technically optimize by using the output buffer for absorbing + squeezing instead of an internal one which is later copied?